src/test/java/edu/ucla/library/bucketeer/verticles/ClearCacheIT.java

```diff
@@ -0,0 +1,365 @@
+            myVertID = verticleID;
+            return sendMessage(imageIDNull);
+        }).onFailure(failure -> {
+            if (Integer.parseInt(failure.getMessage()) == HTTP.INTERNAL_SERVER_ERROR) {
+                TestUtils.complete(asyncTask);
+            } else {
+                aContext.fail();
+            }
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticle deployment fails when given null credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testNullCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject nullConfigs = myConfigs;
+
+        nullConfigs.put(IIIF_USERNAME, (String) null);
+        nullConfigs.put(IIIF_PASSWORD, (String) null);
+
+        deployNewVerticle(nullConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticle deployment fails when given invalid credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testInvalidCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject newConfigs = myConfigs;
+
+        newConfigs.put(IIIF_USERNAME, "username");
+        newConfigs.put(IIIF_PASSWORD, "password");
+
+        deployNewVerticle(newConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Gets width from info.json from Cantaloupe of specific image
+     */
+    private Future<Integer> getImageWidth() {
+        final WebClient client = WebClient.create(Vertx.vertx());
+        final Promise<Integer> promise = Promise.promise();
+
+        LOGGER.debug(myIIIfURL + IIIF_RESOURCE);
+
+        client.getAbs(myIIIfURL + IIIF_RESOURCE).putHeader(CONTENT_TYPE, JSON).send(request -> {
+            if (request.succeeded()) {
+                final HttpResponse<Buffer> response = request.result();
+
+                if (response.statusCode() == HTTP.OK) {
+                    promise.complete(response.bodyAsJsonObject().getInteger("width"));
+                } else {
+                    promise.fail(LOGGER.getMessage(MessageCodes.BUCKETEER_611, response.statusCode(),
+                            response.statusMessage()));
+                }
+            } else {
+                promise.fail(request.cause());
+            }
+
+            client.close();
+        });
+
+        return promise.future();
+    }
+
+    /**
+     * Puts image into AmazonS3 bucket
+     */
+    private Future<Void> putObjectS3(final String aPath) {
+        final Promise<Void> promise = Promise.promise();
+
+        try {
+            myAmazonS3.putObject(s3Bucket, IMAGE_JPX, new File(aPath));
```

Comment:
Is this (line 318) a synchronous call? If so, it doesn't seem like you really need a method for this? Why not just call myAmazonS3.putObject in the main code? Is this just to isolate the AmazonServiceException try/catch? The method could be synchronous if that method is?

Reply:
It is an asynchronous call. putObject() can take a few seconds sometimes, and we needed to use the promise to guarantee the object was put in before trying to access anything from Cantaloupe.